examples/data_parallel/train_resnet_ddp.py

```diff
@@ -20,6 +20,28 @@ def __init__(self):
     self.model = DDP(
         self.model, gradient_as_bucket_view=True, broadcast_buffers=False)
     self.optimizer = optim.SGD(self.model.parameters(), weight_decay=1e-4)
+    # below code is commented out because in this example we used a fake data
+    # loader that does not take sampler. However this logic is needed if you
+    # want each process to handle different parts of the data.
+    '''
```

Comment:
Is it clearer to just apply this sampler to the fake dataset anyway?

Reply:
fake dataset is a XLA util https://github.com/pytorch/xla/blob/master/examples/train_resnet_base.py#L25-L29, it does not take sampler.

Reply:
Oh, I just kind of assumed that `SampleGenerator` was an idiomatic `Dataset`. Can you try just making it inherit from [`IterableDataset`](https://github.com/pytorch/xla/blob/00e406325f5c7c0897680dcb38ca6b168ce165ed/torch_xla/utils/utils.py#L56-L57) since it has `__iter__` and `__len__` already? You should then be able to wrap it in a standard sampler and data loader.

If that doesn't work, then one of us can follow up. Our examples should be as close to PyTorch as possible.